crates/sui-data-ingestion/src/progress_store/mod.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+mod dynamodb;
+pub use dynamodb::DynamoDBProgressStore;
+
+pub type ExecutorProgress = HashMap<String, CheckpointSequenceNumber>;
+
+#[async_trait]
+pub trait ProgressStore: Send {
```

Comment:
Is an s3/file based progress store also in the works? Something which is also possible for non aws users?

Reply:
sure, that's why it's a trait. The interface is quite simple to implement. We can also add a file-based progress store to default built-ins for people to use

Reply:
added a file-based progress store and updated tests to utilize it